setup.py

```diff
@@ -256,9 +256,11 @@ def finalize_options(self):
         Extension('gensim.models.doc2vec_inner',
             sources=['./gensim/models/doc2vec_inner.c'],
             include_dirs=[model_dir]),
+        Extension('gensim.corpora.mmreader',
+            sources=['./gensim/corpora/mmreader.c']),
```

Comment:
Probably, "include_dirs" should be `gensim/corpora`

Reply:
Thanks @menshikh-iv .  It looks like it is failing on the bold line below in the tox config:

<pre>
<b>python -c "from gensim.models.word2vec import FAST_VERSION; print(FAST_VERSION)"</b>
python setup.py build_ext --inplace
python -c "from gensim.models.word2vec import FAST_VERSION; print(FAST_VERSION)"
</pre>

This tries to import word2vec before any of the cython modules are compiled.  word2vec  imports gensim.corpor, which import mmreader.  Since I took out the python version of mmreader, this line is failing as the cython version isn't built yet (this happens on the next line of the tox config).

For now, I removed the bold line to confirm the build works.  If the plan is to only include the c/cython versions of code, I am not sure if the bold line above is still necessary.  If it is still necessary, then I can add a try/except to gensim.corpora.__init__, though that would only be used for this line in the tox file.

Reply:
@arlenk aha, this means that now we can't use `FAST_VERSION` freely, this isn't good. 

Probably, we need to move this constant (because this needed for checking that all compiled successfully), to a zone where we didn't receive an exception if something goes wrong.

Reply:
Ok, I added back the FAST_VERSION check to the tox config, so the config is the same as before.

To make the test run, I had to add a try/except block to gensim.corpora.__init__ as that file tries to import MmCorpus.  Since this check happens before the c-extension is built, MmCorpus fails because there is no pure python MmReader any more.  The except block in the __init__ file just creates a dummy MmCorpus class so the code "works" (ie, the imports don't fail, but obviously the dummy classes will not work if used).

This is not a very elegant solution, but I can't think of a better option if that check in the tox file is necessary.  Maybe there is a different way to check if the compilation step failed?  Also, is the compilation step still necessary with a wheel distribution?

Reply:
Yes, this is not a good solution.
Compilation step is a **mandatory** step for wheels, I agree, but we also have repo & developers who work with it.

As alternative, I propose to replace import in `tox.ini` to something like this `from gensim.models.word2vec_inner import FAST_VERSION`, i.e. direct import, this will be helpful for this case?

Reply:
> As alternative, I propose to replace import in tox.ini to something like this from gensim.models.word2vec_inner import FAST_VERSION, i.e. direct import, this will be helpful for this case?

That's a good idea, but unfortunately that fails too.  Importing anything under gensim.* implicitly runs gensim/__init__.py, which imports gensim.corpora, which imports mmreader (which doesn't exist yet at this point in the tox process).

Do you know what this line in the tox config is meant to test?

<pre>
<b>python -c "from gensim.models.word2vec import FAST_VERSION; print(FAST_VERSION)"</b>
python setup.py build_ext --inplace
python -c "from gensim.models.word2vec import FAST_VERSION; print(FAST_VERSION)"
pytest {posargs:gensim/test}
</pre>

It seems like it tests whether all the "pure python" code installed correctly, but I wonder if there is a different way to test that.  Or is that test necessary since either the following line (the setup.py line) or the pytest line will fail if the install was not good.

Reply:
> Do you know what this line in the tox config is meant to test?

I added it when I write the first version of `tox.ini` for sanity checking only

Reply:
Feel free to remove this (and also remove try/except block in python file)

Reply:
> Do you know what this line in the tox config is meant to test?
> 
> I added it when I write the first version of tox.ini for sanity checking only

ah, ok, that makes sense.  I will remove this line for now and drop the try/except. 

thanks